test/export/test_export.py

```diff
@@ -109,7 +109,163 @@ def f(x, y):
         inp = ([torch.ones(1, 3)], torch.ones(1, 3))
         self._test_export_same_as_eager(f, inp)
 
-    def test_export_preserve_signature(self):
+    def compare_outputs(self, eager, unflattened, args):
+        orig_output = eager(*args)
+        unflattened_output = unflattened(*args)
+        self.assertTrue(torch.allclose(orig_output, unflattened_output))
+
+    def test_unflatten_nested(self):
```

Comment:
can we make this a separate test file? The unflattener is a logically separate component from export, and it would be nice to avoid bloating this file over time.